server/midddleware/request_metadata.go

```diff
@@ -0,0 +1,101 @@
+package middleware
+
+import (
+	"context"
+
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+type RequestMetadataCtxKey struct {
+}
+
+type AccessToken struct {
+	Namespace string
+	Sub       string
+}
+
+type RequestMetadata struct {
+	accessToken AccessToken
+	namespace   metadata.Namespace
+}
+
+// NamespaceExtractor - extract the namespace from context
+type NamespaceExtractor interface {
+	Extract(ctx context.Context) (metadata.Namespace, error)
+}
+
+type AccessTokenNamespaceExtractor struct {
+	TenantManager      *metadata.TenantManager
+	Transactionmanager *transaction.Manager
+}
+
+func GetRequestMetadata(ctx context.Context) (*RequestMetadata, error) {
+	// read token
+	value := ctx.Value(RequestMetadataCtxKey{})
+	if value != nil {
+		if requestMetadata, ok := value.(*RequestMetadata); ok {
+			return requestMetadata, nil
+		}
+	}
+	return nil, api.Errorf(api.Code_NOT_FOUND, "request metadata not found 1")
```

Comment:
any reason of adding `1` in the error string?

Reply:
fixed. (it was left over from debugging)